src/llm/apis/openai_completions.cpp

```diff
@@ -869,17 +881,36 @@ std::string OpenAIChatCompletionsHandler::serializeUnaryResponse(const std::vect
             writer.StartObject();  // {
             // content: string; Actual content of the text produced
             writer.String("content");
-            writer.String(completeResponse.c_str());
+            writer.String(parsedResponse.content.c_str());
             // role: string; Role of the text producer
             // Will make sense once we have chat templates? TODO(atobisze)
             writer.String("role");
             writer.String("assistant");  // TODO - hardcoded
-            // TODO: tools_call
-            // TODO: function_call (deprecated)
+
+            writer.String("tool_calls");
+            writer.StartArray();  // [
+            for (const ToolCall& toolCall : parsedResponse.toolCalls) {
+                writer.StartObject();  // {
+                writer.String("id");
+                writer.String(toolCall.id.c_str());  // Generate a random ID for the tool call
+
+                writer.String("type");
+                writer.String("function");
+
+                writer.String("function");
+                writer.StartObject();  // {
+                writer.String("name");
+                writer.String(toolCall.name.c_str());
+                writer.String("arguments");
+                writer.String(toolCall.arguments.c_str());  // Assuming toolsResponse is a valid JSON string
```

Comment:
is it validated anywhere?

Reply:
Yes. Check out last phases of parsing the the parse() method.